profiles/seccomp/seccomp_default.go

```diff
@@ -1297,7 +1297,15 @@ func DefaultProfile(rs *specs.Spec) *types.Seccomp {
 		{
 			Name:   "setsockopt",
 			Action: types.ActAllow,
-			Args:   []*types.Arg{},
+			Args: []*types.Arg{
+				{
+					// disallow IPT_SO_SET_REPLACE, IP6T_SO_SET_REPLACE, ARPT_SO_SET_REPLACE
+					Index:    2,
```

Comment:
kk will update

On Tue, Jul 5, 2016 at 4:25 AM, Justin Cormack <notifications@github.com>
wrote:

> In profiles/seccomp/seccomp_default.go
> <https://github.com/docker/docker/pull/23893#discussion_r69546695>:
>
> > @@ -1297,7 +1297,15 @@ func DefaultProfile(rs *specs.Spec) *types.Seccomp {
> >  		{
> >  			Name:   "setsockopt",
> >  			Action: types.ActAllow,
> > -			Args:   []*types.Arg{},
> > +			Args: []*types.Arg{
> > +				{
> > +					// disallow IPT_SO_SET_REPLACE, IP6T_SO_SET_REPLACE, ARPT_SO_SET_REPLACE
> > +					Index:    2,
>
> We should only disallow IPT_SO_SET_REPLACE and ARPT_SO_SET_REPLACE when
> arg 1 (level) is SOL_IP (0) and IP6T_SO_SET_REPLACE when arg 1 is SOL_IPV6
> (41), otherwise we might be blocking valid socket options on different
> address families.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/docker/pull/23893/files/32c532c6c1d98735e604fa3aee36a6c67c29b608#r69546695>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABYNbNUluby0bnhXCYtjMzpI40vR9hVjks5qSj8RgaJpZM4I8hV5>
> .
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>